openmdao/solvers/newton.py

```diff
@@ -116,6 +121,7 @@ def solve(self, params, unknowns, resids, system, metadata=None):
         system.children_solve_nonlinear(local_meta)
         system.apply_nonlinear(params, unknowns, resids)
 
+        unknowns_cache = np.zeros(unknowns.vec.shape)
```

Comment:
Might be better to create unknowns_cache in the setup() method so we don't reallocate a new one for every call to solve().  Also, when we do create it, we could use np.empty() instead of np.zeros() since we're replacing all of the values later, so no need to waste time initializing the whole vector to zeros to start with.